### PR TITLE
Adamdebek/fs mark

### PR DESCRIPTION
**Pull request by adamdebek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: fs_mark/build.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+PREFIX_FS_MARK_SRC="${PREFIX_FS_MARK_BUILD}/${FS_MARK}"
+PREFIX_FS_MARK_MARKERS="${PREFIX_FS_MARK_BUILD}/markers"
+
+#
+# Download and unpack
+#
+mkdir -p "$PREFIX_FS_MARK_BUILD" "$PREFIX_FS_MARK_MARKERS"
+if ! [ -f "${PREFIX_FS_MARK}/${FS_MARK}.tar.gz" ]; then
+	if ! wget "$PKG_URL" -O "${PREFIX_FS_MARK}/${FS_MARK}.tar.gz" --no-check-certificate; then
+		wget "$PKG_MIRROR_URL" -P "${PREFIX_MBEDTLS}" --no-check-certificate
```
Comment (nalajcie) [inline fs_mark/build.sh:22]: invalid prefix (MBEDTLS)

File: fs_mark/build.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+set -e
+
+FS_MARK_VER="3.3"
+FS_MARK=fs_mark-${FS_MARK_VER}
+PKG_URL="https://github.com/josefbacik/fs_mark/archive/master.tar.gz"
```
Comment (nalajcie) [inline fs_mark/build.sh:7]: you're downloading current master branch, not explicit version 3.3 - I see that there is no v3.3 tag/branch and the actual code haven't changed in years - but maybe downloading explicit commit would be better (right now upstream commit might break our port)

https://github.com/josefbacik/fs_mark/archive/2628be58146de63a13260ff64550f84275556c0e.zip

File: fs_mark/patch/fs_mark.patch [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+-#include <linux/unistd.h>
++#include <limits.h>
++#include <signal.h>
+ 
+ #include "fs_mark.h"
+ 
+ void cleanup_exit(void)
+ {
+-	char child_log_file_name[PATH_MAX];
++	char child_log_file_name[PATH_MAX + 16];
```
Comment (nalajcie) [inline fs_mark/patch/fs_mark.patch:28]: why is this change needed? IMHO our OS won't support paths larger than PATH_MAX (as the name suggest :p) so there is no point to create larger buffers

**Review by nalajcie — CHANGES_REQUESTED** [review]
Pointed out some issues. 

Please also remove the other commit which got here by accident

File: fs_mark/patch/fs_mark.patch (hunk 3 above)
Comment (adamdebek) [reply fs_mark/patch/fs_mark.patch:28]: I did to get rid of warinings caused by:
```C
sprintf(child_log_file_name, "%s.%d", log_file_name, child_pid);
```
since log_file_name is defined as char[PATH_MAX], compilator complains that there could be overflow,
compilator says that 11 byes will be sufficient so I added 16 (power of 2)

File: fs_mark/build.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -e
+
+FS_MARK_VER="3.3"
+FS_MARK=fs_mark-${FS_MARK_VER}
+FS_MARK_COMMIT="2628be58146de63a13260ff64550f84275556c0e"
+PKG_URL="https://github.com/josefbacik/fs_mark/archive/2628be58146de63a13260ff64550f84275556c0e.tar.gz"
```
Comment (nalajcie) [inline fs_mark/build.sh:8]:
```suggestion
PKG_URL="https://github.com/josefbacik/fs_mark/archive/${FS_MARK_COMMIT}.tar.gz"
```

File: fs_mark/patch/fs_mark_h.patch [hunk 5]
```diff
@@ -0,0 +1,11 @@
+--- fs_mark-3.3/fs_mark.h	2023-07-10 15:47:05.889696230 +0200
++++ fs_mark-3.3.phoenix/fs_mark.h	2023-07-31 12:29:45.037726594 +0200
+@@ -125,7 +125,7 @@
+ int	num_threads = 1;			/* Number of threads */
+ int	do_fill_fs = 0;				/* Run until the file system is full  */
+ int	verbose_stats = 0;		    	/* Print complete stats for each system call */
+-char 	log_file_name[PATH_MAX] = "fs_log.txt"; /* Log file name for run */
++char 	log_file_name[PATH_MAX - 16] = "fs_log.txt"; /* Log file name for run */
```
Comment (nalajcie) [inline fs_mark/patch/fs_mark_h.patch:8]:
```suggestion
+char 	log_file_name[PATH_MAX - 16] = "fs_log.txt"; /* Log file name for run - reserve place for the suffix */
```

**Review by nalajcie — COMMENTED** [review]
small nitpicks, otherwise LGTM

**Review by nalajcie — APPROVED** [review]

**Review by agkaminski — APPROVED** [review]
LGTM

**Review by damianloew — APPROVED** [review]
